docs/js/features/odata/generate-client.md

```diff
@@ -0,0 +1,115 @@
+---
+id: generator-js-sdk
+title: Generate a typed OData client for JavaScript
+hide_title: false
+hide_table_of_contents: false
+sidebar_label: Generate an OData client for JavaScript
+keywords:
+- sap
+- cloud
+- sdk
+- odata
+- JavaScript 
+- TypeScript
+- generate
+author: Charles Dubois
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Generate a typed OData client with the OData Generator
+
+The generator allows you to generate a custom OData client library for any service. You can then easily access these services via the client libraries.
+
+There are two way to use the generator:
+- Via the command line interface (CLI)
+- By instantiating and invoking it at runtime
+
+For any of them you need to have an `EDMX` file holding the service metadata.
+
+## Installation
+
+Run this command in your project's terminal to install the generator globally.
+```sh
+npm install -g @sap-cloud-sdk/generator
+```
+
+:::note
+The generator is included in the SAP cloud-sdk-cli, if you have it installed you don't need to install the generator.
+:::
+
+## Usage
+
+### Using the CLI
+
+The generator is primarily meant to be used on the command line:
+
+```sh
+generate-odata-client --inputDir path/to/your/service-specification(s) --outputDir path/where/the/modules/are/stored
+```
+
+Run `generate-odata-client --help` for further options.
+
+### Invoke the generator programmatically
+
+You can also use the generator programmatically. You will have to provide the options anyways.
+
+1. Import the generator in your project.
+```ts
+import { generateProject } from '@sap-cloud-sdk/generator';
+```
+
+2. Create your options, adapt the input & output directory as well as the package name according to your setup.
+```ts
+const options: GeneratorOptions = initializeOptions();
```

Comment:
`initializeOptions` is not defined anywhere.

Reply:
I don't know how to use the generator in typescript, only code I found was https://github.com/SAP/cloud-sdk/blob/master/packages/generator/README.md

Reply:
If I could get a code snippet showing the generator in typescript that would be great

Reply:
Makes sense. The generator unfortunately need way more options to be invoked programatically then when you use it on the CLI. Take a look at the [generate test services script](https://github.com/SAP/cloud-sdk/blob/10621361dc106ff5e05e1c92e55ce871c8fe2d59/scripts/generate-test-services.ts#L41). There we use the generator to generate our own test services (and do some more stuff. I would recommend to play around with it and find out which are the mandatory parameters (as I said, there are a lot of them unfortunately - we will fix that at some point I hope).